app/page/authorization/useUserOperation.ts

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState } from "react"
+import { useClsState } from "use-cls-state"
+import { useHashLocation } from "wouter/use-hash-location"
+
+import { useProviderContext } from "~app/context/provider"
+import { Path } from "~app/path"
+import { useAction } from "~app/storage"
+import type { Account } from "~packages/account"
+import {
+  UserOperationV0_6,
+  UserOperationV0_7,
+  type UserOperation
+} from "~packages/bundler/userOperation"
+import { WaalletRpcMethod } from "~packages/waallet/rpc"
+import {
+  TransactionType,
+  type Network,
+  type TransactionPending
+} from "~storage/local/state"
+
+import type { PaymentOption } from "./transaction"
+
+export function useUserOperation(props: {
```

Comment:
I feel this hook might not be an ideal extraction, since it is doing too many things (navigation, payment, gas estimation, user op action), that developer might not be aware all of them. 

I would suggest we could inline them as before (including `usePayment`), and discuss for a better modularization.

Reply:
I agree with you. Maybe we could discuss it later.
(The reason I would like move these logic outside is that `UserOperationConfirmation` handles too many things, such as user operation, payment and markup.)